### ISSUE TITLE
does not distinguish type hint and type alias

### ISSUE THREAD

**Issue by yuyupopo** [python/mypy#10236]
**Bug Report**

mypy seems to think a class type hint is a type alias.

**To Reproduce**

```python
# example.py

class A:
    pass

class B:
    A: int
    a: A  # mypy error: Variable "example.B.A" is not valid as a type

# mypy example.py
```

**Expected Behavior**

I expected no errors.

**Actual Behavior**

```
example.py:6: error: Variable "ex.B.A" is not valid as a type
example.py:6: note: See https://mypy.readthedocs.io/en/latest/common_issues.html#variables-vs-type-aliases
```

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.7.10
- Operating system and version: macOS 10.15.7

**Comment by DevilXD** [comment]
This may be because of how the class body is built. When the methods are defined, the class' body is executed in a separate scope. This means that referencing `A` points to the local scope first (the class var in this case), and only if that wouldn't exist in the class' body, then it'd search for it in the outer scopes. Here's a section from Python docs on this: https://docs.python.org/3/reference/datamodel.html#executing-the-class-body

I've ran into this issue myself while working with discord.py, and haven't been able to find any good way around it, other than just using different var / func / class names. Here's a simplified version of what happened in my case: https://i.imgur.com/s1e9BLT.png
Trying to parametrize the `set` typing in this case, would result in MyPy telling you that it's a method, not a type.

Your truly best bet, is to not use `A` as the name of the class, or as the name of the class var inside `B`s definition.

**Comment by yuyupopo** [comment]
While I agree that the best practice is to not use `A` as an attribute name, I still see this as a bug.
In your example, `set` is a method understood by the python interpreter. 
In my example, `A` is a type annotation, and cannot be interpreted in the runtime. (`B.A`)

**Comment by DevilXD** [comment]
@yuyupopo In my example tho, I'm also using a type annotation, in the `func` function - the issue is that `set` refers to the method of the defined class, instead of the built-in `set` class. This is basically the same issue as you're having, and an intended base Python behavior at the current point in time. MyPy has nothing to do with this here, unless you're implying MyPy should make an exception for this particular case (and explicitly search in the outer scopes first), in which case I don't agree it's something worth adding, considering it wouldn't match the standard Python behavior.

**Comment by hauntsaninja** [comment]
On master you can use PEP 613 for this kind of thing `A: TypeAlias = int`
